docs/authoring.md

```diff
@@ -16,13 +16,52 @@ The AsyncAPI generator has been built with extensibility in mind. The package us
 1. The default variables you have access to in any the template file are the following:
    - `asyncapi` that is a parsed spec file object. Read the [API](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument) of the Parser to understand to what structure you have access in this parameter.
    - `originalAsyncAPI` that is an original spec file before it is parsed. 
-1. To separate functionality into files, you can specify a filename like `$$channel$$.js` to generate a file for each channel defined in your AsyncAPI document. Such files templating gives you access to the following extra variables:
-   - When using `$$channel$$` you have access to two variables `channel` and `channelName`. Where `channel` contains the current channel being rendered.
-   - When using `$$message$$` you have access to two variables `message` and `messageName`. Where `message` contains the current message being rendered.
-   - When using `$$schema$$` you have access to two variables `schema` and `schemaName`. Where `schema` contains the current schema being rendered.
-   - When using `$$parameter$$` you have access to two variables `parameter` and `parameterName`. Where `parameter` contains the current parameter being rendered.
-   - When using `$$securityScheme$$` you have access to two variables `securityScheme` and `securitySchemeName`. Where `securityScheme` contains the current security scheme being rendered.
-   
+## File templates
+
+It is possible to generate files for each specific object in your AsyncAPI documentation. 
+For example, you can specify a filename like `$$channel$$.js` to generate a file for each channel defined in your AsyncAPI. 
+The following file-template names and extra variables in them are available:
+   - `$$channel$$`, within the template-file you have access to two variables [`channel`](https://github.com/asyncapi/parser-js/blob/master/API.md#Channel) and [`channelName`](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument+channels). Where `channel` contains the current channel being rendered.
```

Comment:
```suggestion
   - `$$channel$$`, within the template-file you have access to two variables [`channel`](https://github.com/asyncapi/parser-js/blob/master/API.md#Channel) and `channelName`. Where `channel` contains the current channel being rendered and `channelName` is the channel path to the corresponding channel. 
```
@Tenischev how about something like this? 🤔

Reply:
@jonaslagoni sorry, couldn't agree. The 

> `channelName` is the channel path to the corresponding channel

is super abstract for me. And also i doubt about "channel path" phrase, this could be just name of topic, name of queue e.t.c.

Reply:
What about following?
`$$channel$$` iterates over all channels. Within the template-file you have access to two variables `channel` and `channelName`, which are defined as `{% channel, channelName in asyncapi.`[`channels()`](https://github.com/asyncapi/parser-js/blob/master/API.md#AsyncAPIDocument+channels)` %}`.

Reply:
IMO this gives no clarity to what `channelName` is, but for me it is pretty self-explanatory what channel path is when having read the specification, but yea since I am already somewhat involved with the parser and generator, it is really hard for me to say x, y, z is easier to understand as newcomers. So unless someone else weighs in i'd say you make the final call 😄